### PR TITLE
slack-bot: handle interactions by filing issues

### PR DESCRIPTION
**Pull request by stevekuznetsov**
With this patch, the test platform Slack integration now knows how to
respond to a number of shortcut actions from users, opening dialogs that
prompt them for information in order to open issues on the DPTP Jira
board. The following types of issues are supported:
 - bugs
 - consulting requests
 - feature requests
 - incident tracking cards

Input validation is achieved with custom validation on submissiosn as
well as built-in Slack modal features; users have interactive flows to
provide them feedback on the inputs and allow them to open bugs against
this tool if we run into unexpected issues.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
```
$ make update-vendor 
docker run --rm \
	--user=$UID \
	-v $(go env GOCACHE):/.cache:Z \
	-v $PWD:/go/src/github.com/openshift/ci-tools:Z \
	-w /go/src/github.com/openshift/ci-tools \
	-e GO111MODULE=on \
	-e GOPROXY=https://proxy.golang.org \
	golang:1.14 \
	/bin/bash -c "go mod tidy && go mod vendor"
Trying to pull registry.fedoraproject.org/golang:1.14...
  manifest unknown: manifest unknown
Trying to pull registry.access.redhat.com/golang:1.14...
  name unknown: Repo not found
Trying to pull registry.centos.org/golang:1.14...
  manifest unknown: manifest unknown
Trying to pull docker.io/library/golang:1.14...
Getting image source signatures
Copying blob 57df1a1f1ad8 done  
Copying blob 03f1c9932170 done  
Copying blob 71e126169501 done  
Copying blob f4773b341423 done  
Copying blob 1af28a55c3f3 done  
Copying blob 6e11e1e4650d done  
Copying blob 7bdfcd218115 done  
Copying config d6747a1383 done  
Writing manifest to image destination
Storing signatures
build cache is required, but could not be located: GOCACHE is not defined and neither $XDG_CACHE_HOME nor $HOME are defined
make: *** [Makefile:89: update-vendor] Error 1
```

@alvaroaleman do you know how this is supposed to work?

File: cmd/slack-bot/main_test.go
Comment (alvaroaleman) [inline cmd/slack-bot/main_test.go:1]: :trollface:

**Comment by stevekuznetsov** [comment]
This implements DPTP-1476

**Comment by bbguimaraes** [comment]
> Would you mind adding a README with a short architectural description? It could be as simple as a list of links to the relevant API documentation if it's good.

Posting the reply here for other reviewers: https://api.slack.com/surfaces/modals/using.

File: pkg/slack/modals/bug/view_test.go
Comment (petr-muller) [inline pkg/slack/modals/bug/view_test.go:28]: Ugh. How useful will these be in practice?

Comment (petr-muller) [reply pkg/slack/modals/bug/view_test.go:28]: I mean, if/when I need these tests to understand how the input affects the output, I am screwed, because I have no idea how to find the _valid because specific component chosen_ in that JSON. Maybe we should build these inputs dynamically from a template if possible?

**Review by petr-muller — COMMENTED** [review]
I haven't gone line-by-line (surprise!) but the structure makes sense to me and I think I would be able to maintain and extend this if needed, so LGTM. 

One thing that I did not like though are the tests with huge JSON blobs as inputs, I don't think they serve their purpose well, although I see how it's not really easy to make the inputs trivial and easy to read. Could we have some builder for them? And if not, I think we should at least use fixtures.

**Comment by stevekuznetsov** [comment]
All tests use fixtures, responded to all comments. @alvaroaleman @petr-muller

**Review by petr-muller — APPROVED** [review]
:heavy_check_mark:

**Comment by petr-muller** [comment]
/hold 

Fol Alvaro to have a chance

**Pull request merged**
